src/client/components/bubble-menu/bubble-menu.tsx

```diff
@@ -51,6 +52,49 @@ export interface BubbleMenuState {
   y?: number;
 }
 
+interface Coordinates {
+  x: number;
+  y: number;
+}
+
+interface PositionStyles {
```

Comment:
Are those really Styles? These seems more like Attributes to me
What mostly bothers me about Styles is that you are calculating coordinates below and return them as styles

Reply:
I treat `Coordinates` as cardinal/cartesian coordinates. Here, it's styles because position is expressed as subset of css styles attributes and has css semantics. It's not coordinates for sure, but I admit, name's not perfect.

Reply:
In short, coordinates are used to calculate origin point on screen where tooltip should be presented. But from this coordinates you need to calculate proper css attributes/styles taking alignment and direction into consideration. That's why I have two, distinct types.